### PR TITLE
feat(CON-7852): add LB admission webhook integration tests

### PR DESCRIPTION
**Pull request by prichrd**
This change adds integration tests for the admission webhook introduced in #676. By leveraging envtest, we can test the flow of creation and update with the Kubernetes API server.

### CODE REVIEW

**Review by varshavaradarajan — CHANGES_REQUESTED** [review]

File: envtest/lb_admission_test.go [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+
+func TestLBAdmissionWebhook(t *testing.T) {
+	testCases := []struct {
+		name            string
+		givenService    *corev1.Service
+		givenLBResponse *stubLBResponse
+		allow           bool
+	}{
+		{
+			"allowed when service type is not LB",
```
[the remark this answers is not in the document]
Comment (prichrd) [reply envtest/lb_admission_test.go]: It indeed makes it a bit more readable.

File: Makefile [hunk 2]
```diff
@@ -105,10 +107,19 @@ gofmt: # run in script cause gofmt will exit 0 even if files need formatting
...
+
+.PHONY: test-unit
+test-unit:
 	@echo "==> Testing all packages"
 	@GO111MODULE=on GOFLAGS=-mod=vendor go test -race $(shell go list ./... | grep -v vendor)
 
+.PHONY: test-integration
+test-integration:
+	@echo "==> Running integration tests"
+	@command -v setup-envtest &>/dev/null || GOFLAGS="" go install sigs.k8s.io/controller-runtime/tools/setup-envtest@latest
```
[the remark this answers is not in the document]
Comment (prichrd) [reply Makefile:120]: That's what I initially thought.

The tool will download the right set of binaries when calling `setup-envtest use -p env ${ENVTEST_K8S_VERSION}`.

**Comment by varshavaradarajan** [comment]
@prichrd - I'm assuming we will be adding this to the GH actions test workflow in a followup PR?

**Review by varshavaradarajan — APPROVED** [review]

**Comment by prichrd** [comment]
@varshavaradarajan with the [changes to the Makefile](https://github.com/digitalocean/digitalocean-cloud-controller-manager/pull/682/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R110) running `make ci` will run the unit and integration tests (through the `test` target). The tests ran for the current PR: https://github.com/digitalocean/digitalocean-cloud-controller-manager/actions/runs/7716714800/job/21034182439#step:4:12

**Pull request merged**